cpu/stm32/Makefile.include

```diff
@@ -3,6 +3,11 @@ CFLAGS += -DCPU_FAM_STM32$(call uppercase_and_underscore,$(CPU_FAM))
 # For stm32 cpu's we use the generic stm32.ld linker script
 LINKER_SCRIPT ?= stm32.ld
 
+# STM32MP1 Engineering mode (Cortex-M4 only)
+ifeq ($(STM32MP1_ENGINEERING_MODE),1)
+  CFLAGS += -DSTM32MP1_ENGINEERING_MODE
+endif
```

Comment:
Instead of using a CFLAGS, you could also use a pseudo-module. I would find this cleaner. It also needs to be documented, maybe in the board documentation at first.